pkg/codegen/java/templates_gradle.go

```diff
@@ -116,15 +118,19 @@ func newGradleTemplateContext(
 		ctx.LicenceURL = "http://www.apache.org/licenses/LICENSE-2.0.txt"
 	}
 
+	ctx.GroupID = packageInfo.BasePackage
```

Comment:
// The prefix of the Java package to generate.
	//
	// Defaults to "com.pulumi".
	//
	// The generated Java code will start with:
	//
	//     package ${BasePackage}.${pkg.Name};
	//
	// Note that pkg.Name is coming from the Schema.
	//
	// For example, if the package is named "azuread" then the
	// generated code will be placed under:
	//
	//     package com.pulumi.azuread
	//
	// And if BasePackage is set to "com.myorg" then the generated
	// code will instead be placed under:
	//
	//     package com.myorg.azuread
	//
	BasePackage string `json:"basePackage"`

Hm, It's probably reasonable to default GroupID to BasePackage